libr/bin/p/bin_hunk.c

```diff
@@ -84,17 +61,29 @@ static RList* sections(RBinFile *bf) {
 	return ret;
 }
 
-static RList* entries(RBinFile *bf) { //Should be 3 offsets pointed by NMI, RESET, IRQ after mapping && default = 1st CHR
+static RList* entries(RBinFile *bf) {
 	RList *ret;
-	RBinAddr *ptr = NULL;
 	if (!(ret = r_list_new ())) {
 		return NULL;
 	}
+	RBinAddr *ptr = NULL;
 	if (!(ptr = R_NEW0 (RBinAddr))) {
 		return ret;
 	}
-	ptr->paddr = 0x24;
-	ptr->vaddr = 0x24;
+	int addr;
+	ut8 b[1024];
+	int last = r_buf_read_at (bf->buf, 0, b, sizeof (b)) - 4;
+	for (addr = 0x18; addr <= last; addr += 4) {
+		if (!memcmp (b + addr, HUNK_CODE, 4)) {
+			ptr->paddr = addr + 8;
+			ptr->vaddr = addr + 8;
+			r_list_append (ret, ptr);
+			return ret;
+		}
+	}
+	R_LOG_ERROR ("Cannot determine entrypoint, cannot find HUNK_CODE");
+	ptr->paddr = 0;
```

Comment:
Free ptr and return null. Im fine with the error message but if theres no entry. Handling nullable return value here should be better than asumimg address zero

Reply:
lets return empoty list, see comment below.